rusoto/core/src/request.rs

```diff
@@ -35,14 +38,63 @@ lazy_static! {
             env!("CARGO_PKG_VERSION"), RUST_VERSION, env::consts::OS).as_bytes().to_vec()];
 }
 
+/// HTTP headers
+#[derive(Debug)]
+pub struct Headers(HashMap<Ascii<String>, String>);
```

Comment:
What were your thoughts on going with using `unicase`/`Ascii` here over just converting the strings to lowercase prior to inserting?

Without having thought too deep about this, in terms of performance as well as complexity it seems like just doing `.to_ascii_lowercase()` when creating the map from the iterator wouldn't be too bad either?

Reply:
>>> What were your thoughts on going with using unicase/Ascii here over just converting the strings to lowercase prior to inserting?

I just looked at how hyper stores headers and figured this should work for Rusoto too.

>>> Without having thought too deep about this, in terms of performance as well as complexity it seems like just doing .to_ascii_lowercase() when creating the map from the iterator (and doing the same on .get() -- or alternatively changing codegen to do it on code-generation time) wouldn't be too bad either?

I guess just having lower-case keys is a bit faster and certainly less complex. I'm not to sure about changing codegen. Mandating that lower-case strings are passed to `get()` could break existing code. If it just lead to a compilation error I'd not be to too worried but this is something that could lead to nasty runtime errors.

Reply:
Ah, alright! I was missing that this was inspired by how hyper does it.

I can see your point with mandating lower-case strings, and I'm happy with the way you did the changes!